app/services/hmpps_api/assessment_api.rb

```diff
@@ -0,0 +1,28 @@
+module HmppsApi
+  class AssessmentApi
+    def self.client
+      host = Rails.configuration.assessment_api_host
+      HmppsApi::Client.new(host)
+    end
+
+    def self.get_latest_oasys_date(offender_no)
+      safe_offender_no = URI.encode_www_form_component(offender_no)
+      route = "/offenders/nomisId/#{safe_offender_no}/assessments/summary?assessmentType=LAYER_3&assessmentStatus=COMPLETE"
+
+      return nil if get_all_assessments(route).empty? || get_all_assessments(route).is_a?(Hash) # returns a 404 status offender not found
+
+      Date.parse(returns_most_recent_assessment(get_all_assessments(route)))
+    end
+  end
+end
+
+private
+
+def get_all_assessments(route)
+  self.client.get(route, cache: false)
```

Comment:
Remove `cache: false` (so it defaults to on)

Reply:
Only call this once – save the first GET response as a variable.

If you want to: you can update your tests to enforce that you only perform 1 HTTP GET request.